warning when compiling fread on master on windows rtools42

```r
> remotes::install_github("Rdatatable/data.table@e9a323de01a17af70d5316016606fa8d35b25023")
...
gcc  -I"c:/PROGRA~1/R/R-42~1.0/include" -DNDEBUG     -I"c:/rtools42/x86_64-w64-mingw32.static.posix/include"  -fopenmp   -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c fread.c -o fread.o
In file included from fread.h:3,
                 from fread.c:27:
dt_stdio.h:25: warning: "__USE_MINGW_ANSI_STDIO" redefined
   25 |   #define __USE_MINGW_ANSI_STDIO 1
      | 
In file included from c:\rtools42\x86_64-w64-mingw32.static.posix\include\corecrt.h:10,
                 from c:\rtools42\x86_64-w64-mingw32.static.posix\include\crtdefs.h:10,
                 from c:\rtools42\x86_64-w64-mingw32.static.posix\include\time.h:9,
                 from c:\rtools42\x86_64-w64-mingw32.static.posix\include\sys\time.h:10,
                 from fread.c:8:
c:\rtools42\x86_64-w64-mingw32.static.posix\include\_mingw.h:435: note: this is the location of the previous definition
  435 | #define __USE_MINGW_ANSI_STDIO 0      /* was not defined so it should be 0 */
      | 
```

--- a/src/fread.c
+++ b/src/fread.c
@@ -1,3 +1,5 @@
+#include "fread.h"
+
 #if defined(CLOCK_REALTIME) && !defined(DISABLE_CLOCK_REALTIME)
 #define HAS_CLOCK_REALTIME
 #endif
@@ -24,7 +26,6 @@
   #include <math.h>      // ceil, sqrt, isfinite
 #endif
 #include <stdbool.h>
-#include "fread.h"
 #include "freadLookups.h"
 
 // Private globals to save passing all of them through to highly iterated field processors

--- a/src/fread.c
+++ b/src/fread.c
@@ -1,5 +1,5 @@
 #include "fread.h"
-
+// include fread.h should happen before include time.h to avoid compilation warning on windows about re-defining __USE_MINGW_ANSI_STDIO, PR#5395.
 #if defined(CLOCK_REALTIME) && !defined(DISABLE_CLOCK_REALTIME)
 #define HAS_CLOCK_REALTIME
 #endif

